optuna/samplers/_base.py

```diff
@@ -216,15 +218,15 @@ def reseed_rng(self) -> None:
         same values. To prevent this issue, this method assigns a different seed to each random
         number generator.
         """
+        return
 
-        pass
 
-    def _raise_error_if_multi_objective(self, study: Study) -> None:
-        if study._is_multi_objective():
-            raise ValueError(
-                "If the study is being used for multi-objective optimization, "
-                f"{self.__class__.__name__} cannot be used."
-            )
+def _raise_error_if_multi_objective(sampler: BaseSampler, study: Study) -> None:
```

Comment:
While `_rase_error_if_multi_objective` is private, some user-defined samplers use it, and these will not be compatible with Optuna v4.
https://github.com/Yard1/autopipeline/blob/b3c8c325adb969ddf9585d313f89dc53e0f6c5eb/automl/search/samplers/optuna_rf_sampler.py#L694